### PR TITLE
Make sitemon-js work with sitemon GenericDataProcessor format

### PR DESCRIPTION
**Pull request by mishan**
@djvirgen

### CODE REVIEW

**Comment by mishan** [comment]
Example usage:

     sitemon.send({'method':'GET', 'pattern':'/users/:userId', 'brand':'Tagged'}, {'calls': 1, 'process_time_ms' : 100});

**Comment by mishan** [comment]
@djvirgen

**Comment by djvirgen** [comment]
Very cool, thanks for doing this! The unit tests will have to be updated, but I like that this is overall a better solution to writing by writing to the `GenericDataProcessor` processor.

File: lib/index.js [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+function Sitemon(table, hosts) {
+    this._table = table;
     this._hosts = Array.isArray(hosts) ? hosts : [hosts];
     this._totalHosts = this._hosts.length;
     this._currentHostIndex = 0;
     this._client = dgram.createSocket("udp4");
 }
 
-Sitemon.prototype.send = function(data) {
+Sitemon.prototype.send = function(keys, values) {
```
Comment (djvirgen) [inline lib/index.js]: The argument names `keys` and `values` seem to be misleading. I'm not sure what the variable names should be, but it appears that one is for "user data" and the other is for "meta data" (data about the data).

File: lib/index.js [hunk 2]
```diff
@@ -31,14 +31,17 @@ Sitemon.prototype.send = function(data) {
...
-    var message = ['type=' + type];
+var formatData = function(table, keys, values) {
+    var message = ['table=' + table];
 
-    for (var key in data) {
-        message.push(key + '=' + data[key]);
+    for (var key in keys) {
+        message.push('k.' + key + '=' + keys[key]);
+    }
+    for (var key in values) {
```
Comment (djvirgen) [inline lib/index.js]: `var key` is used twice in this closure. I suggest using a new variable name for the second loop.

**Comment by djvirgen** [comment]
Also, the README should be updated with the new usage.

File: lib/index.js (hunk 1 above)
Comment (mishan) [inline lib/index.js]: These are the nouns used within sitemon. The keys are key-value pairs where the keys are the column names and the values are the keys that get put into the key tables. They are only referenced in the table we log to by a foreign key ID.

The values are values that sitemon will sum together / aggregate per minute. Thus they have to be numbers. The keys again are the column names while the values are integers that will be summed up.

What would you prefer to call them?

**Comment by mishan** [comment]
Updated

File: lib/index.js (hunk 1 above)
Comment (djvirgen) [inline lib/index.js]: OK, that works. Just seems confusing, but if that's what they are in sitemon I guess we should be consistent.

**Comment by djvirgen** [comment]
LGTM! :cherries:

**Pull request merged**